src/api/hooks/storageProviders.ts

```diff
@@ -0,0 +1,58 @@
+import { getRandomIntInclusive } from '@/utils/number'
+import { QueryHookOptions } from '@apollo/client'
+import {
+  GetStorageProviderQuery,
+  GetStorageProvidersQuery,
+  GetStorageProvidersQueryVariables,
+  useGetStorageProviderQuery,
+  useGetStorageProvidersQuery,
+  useGetStorageProvidersCountQuery,
+} from '../queries/__generated__/storageProviders.generated'
+
+type StorageProviderOpts = QueryHookOptions<GetStorageProviderQuery>
+export const useStorageProvider = (id: string, opts?: StorageProviderOpts) => {
+  const { data, ...queryRest } = useGetStorageProviderQuery({
+    ...opts,
+    variables: { where: { id } },
+  })
+  return {
+    storageProvider: data?.storageProviderByUniqueInput,
+    ...queryRest,
+  }
+}
+
+type StorageProvidersOpts = QueryHookOptions<GetStorageProvidersQuery>
+export const useStorageProviders = (variables: GetStorageProvidersQueryVariables, opts?: StorageProvidersOpts) => {
+  const { data, loading, ...rest } = useGetStorageProvidersQuery({ ...opts, variables })
+  return {
+    storageProviders: data?.storageProviders,
+    loading,
+    ...rest,
+  }
+}
+
+export const useStorageProvidersCount = (variables: GetStorageProvidersQueryVariables, opts?: StorageProvidersOpts) => {
+  const { data: connectionData, loading, ...rest } = useGetStorageProvidersCountQuery({ ...opts, variables })
+  return {
+    totalCount: connectionData?.storageProvidersConnection.totalCount,
+    loading,
+    ...rest,
+  }
+}
+
+export const useRandomStorageProviderUrl = () => {
```

Comment:
Maybe we just fetch storage providers list with some higher limit (like 100 or whatever) and just skip fetching the count? Should be simpler